service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -839,6 +839,10 @@ public DynamicList buildDynamicHearingList(UUID selected) {
         return asDynamicList(getHearingDetails(), selected, HearingBooking::toLabel);
     }
 
+    public DynamicList buildDynamicHearingList(List<Element<HearingBooking>> hearingDetails, UUID selected) {
```

Comment:
I know that we do this in multiple places throughout this class but is the class for it? Would it be better in your hearing service?

Reply:
I think in general the Casedata class is huge now. 

I've placed the overloaded methods alongside the others,
```
    public DynamicList buildDynamicHearingList() {
        return buildDynamicHearingList(null);
    }

    public DynamicList buildDynamicHearingList(UUID selected) {
        return asDynamicList(getHearingDetails(), selected, HearingBooking::toLabel);
    }

    public DynamicList buildDynamicHearingList(List<Element<HearingBooking>> hearingDetails, UUID selected) {
        return asDynamicList(hearingDetails, selected, HearingBooking::toLabel);
    }

```